core/os/memory.h

```diff
@@ -41,7 +41,11 @@ constexpr size_t get_aligned_address(size_t p_address, size_t p_alignment) {
 	return (n_bytes_unaligned == 0) ? p_address : (p_address + p_alignment - n_bytes_unaligned);
 }
 
-// Alignment:  ↓ max_align_t        ↓ uint64_t          ↓ max_align_t
+static constexpr size_t MAX_ALIGN = 16; // Note: Using hardcoded value, since on some platforms `alignof(max_align_t)` can be different in different compile units.
+
+static_assert(MAX_ALIGN % alignof(max_align_t) == 0);
```

Comment:
As far as i understand, `max_align_t` being include dependent is a compiler / stdlib bug, right?

If so, I would prefer to limit the fix to the affected system:

```suggestion
#if defined(_WIN32)
// Note: Using hardcoded value, since the value can end up different in different compile units on 32-bit windows
// due to a compiler bug (see GH-113145)
static constexpr size_t MAX_ALIGN = 16;
static_assert(MAX_ALIGN % alignof(max_align_t) == 0);
#else
static constexpr size_t MAX_ALIGN = max_align_t;
#endif
```

Reply:
Yes, seems to be specific to 32-bit GCC (or CRT it's using).